### ISSUE TITLE
Incompatible pointer types

### ISSUE THREAD

**Issue by DavidPotter** [mCodex/react-native-sensitive-info#52]
I'm getting two warnings when building from within Xcode 8.3.3.

* Incompatible pointer types passing 'NSArray *' to parameter of type 'NSError *'
* Incompatible pointer types sending 'const CFStringRef _Nonnull' (aka 'const struct __CFString *const') to parameter of type 'id<NSCopying> _Nonnull'

It appears that these are actually fixed in the repo but because we're using `yarn add react-native-sensitive-info` to add it to our project we aren't getting the latest.  We're getting version 5.1.0, but that was published back in May.

Can you publish a new version that has all the changes from the past 4+ months?

Thanks,
**_David_**

**Comment by mCodex** [comment]
Hello @DavidPotter , I've just released an beta version of RNSensitiveInfo. 😄 

```bash
yarn add react-native-sensitive-info@5.2.0
```

**Comment by DavidPotter** [comment]
Great!  Thanks @mCodex.  That fixed one of the warnings.  I guess I didn't do my homework as there is still one warning left.  It is at line 215, which looks like this:
```swift
reject(@"no_events", @"There were no events", @[[NSNull null]]);
```
The warning is:

> Incompatible pointer types passing 'NSArray *' to parameter of type 'NSError *'

**Comment by mCodex** [comment]
Hi @DavidPotter, thanks for your feedback! It's stranger 🤔 I have to take a look at it, because I have no idea what is happening.

**Comment by DavidPotter** [comment]
I took another look at this today and this is what I learned.  The `reject` function is defined as an instance of type `RCTPromiseRejectBlock` which is defined like this:
```objective-c
typedef void (^RCTPromiseRejectBlock)(NSString *__strong, NSString *__strong, NSError *__strong)
```
The warning is being displayed because an `NSArray` is being passed where an `NSError` is expected.  I'm puzzled (and a little disappointed) as to why this is a warning and not an error.

Wouldn't it be appropriate to just pass a `nil` instead of `@[[NSNull null]]`?
